app/controllers/notes_controller.rb

```diff
@@ -461,4 +448,15 @@ def location
 
     [latitude, longitude]
   end
+
+  def published_notes
+    hidden_nids = Node.hidden_response_nids
+    notes = Node.research_notes.where('node.status = 1')
+
+    if hidden_nids.empty?
+      notes
+    else
+      notes.where('node.nid NOT IN (?)', hidden_nids)
+    end
```

Comment:
notes.where('node.nid NOT IN (?)', hidden_nids)
```suggestion
 notes.where('node.nid NOT IN (?)', hidden_nids)
```

Reply:
I noticed that when the hidden_nids are empty, the query does not return the rest of the notes.
The previous query was `notes.where('node.nid NOT IN (?)', hidden_nids) unless hidden_nids`

Reply:
<img width="634" alt="Screenshot 2020-12-10 at 17 19 56" src="https://user-images.githubusercontent.com/7622875/101783948-ee158980-3b0b-11eb-940d-a223bb9f1928.png">

Reply:
aha - were you able to fix that behavior? I think you're right that it's incorrect. Thanks!

Reply:
Currently, the if..else statement works as expected.. if we put this line back 
`notes.where('node.nid NOT IN (?)', hidden_nids) unless hidden_nids` we'll also have to specify what happens if the hidden_nids are empty.. maybe 
```suggestion
   return  notes.where('node.nid NOT IN (?)', hidden_nids) unless hidden_nids
   return  notes if hidden_nids.empty?
```
Which is still similar to the current code...unless we should change it...

Reply:
Hey @RuthNjeri This looks amazing!! :tada: :clap: 

Just noting here can we also check for the performance for line - 
`notes = Node.research_notes.where('node.status = 1')` and then again querying within this depending on `hidden_nids`. If they aren't empty this query may take more time for execution. 

For handling this problem with `notes.where('node.nid NOT IN (?)', hidden_nids)` can we try this once with `where.not` and see if it works? something like `notes.where.not(node.nid: hidden_nids)`

Reply:
Hi @Tlazypanda Thanks so much for the suggestion! It's makes the code so much cleaner!

I've combined the query you've suggested and the notes one into one line
```suggestion
    Node.research_notes
        .where('node.status = 1')
        .where.not(nid: hidden_nids)
```

I'm currently testing it and it seems that it's working.. pushing it in a few... it might also solve the performance issue of running two queries separately?